src/Microsoft.Health.Checkpoints/Storage/StorageCheckpointClient.cs

```diff
@@ -0,0 +1,143 @@
+
+            try
+            {
+                var resultSegment = _storageClient.GetBlobsAsync(
+                                        traits: BlobTraits.Metadata,
+                                        states: BlobStates.All,
+                                        prefix: $"{partition}/{checkpointIdentifier}",
+                                        cancellationToken: CancellationToken.None)
+                                        .AsPages();
+
+                await foreach (Page<BlobItem> blobPage in resultSegment)
+                {
+                    if (blobPage.Values.Count == 0)
+                    {
+                        _logger.LogWarning($"No blob found for identifier {checkpointIdentifier}");
+                    }
+
+                    if (blobPage.Values.Count > 1)
+                    {
+                        _logger.LogWarning($"Multiple blobs found for identifier {checkpointIdentifier}");
+                    }
+
+                    foreach (BlobItem blobItem in blobPage.Values)
+                    {
+                        DateTimeOffset lastEventTimestamp = DateTime.MinValue;
+
+                        if (blobItem.Metadata.TryGetValue(_lastProcessedDateTime, out var dt))
+                        {
+                            DateTimeOffset.TryParse(dt, null, DateTimeStyles.AssumeUniversal, out lastEventTimestamp);
+                            checkpoint.LastProcessedDateTime = lastEventTimestamp;
+                        }
+
+                        if (blobItem.Metadata.TryGetValue(_lastProcessedIdentifier, out var id))
+                        {
+                            checkpoint.LastProcessedIdentifier = id;
+                        }
+
+                        if (checkpoint.LastProcessedDateTime == DateTime.MinValue || checkpoint.LastProcessedIdentifier == null)
+                        {
+                            _logger.LogWarning($"No valid checkpoint found for {checkpointIdentifier}. Using default checkpoint of {checkpoint.LastProcessedDateTime}");
+                        }
+
+                        checkpoint.ETag = blobItem.Properties.ETag.ToString();
+                        checkpoint.Identifier = checkpointIdentifier;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, ex.Message);
+                throw;
+            }
+
+            return checkpoint;
+        }
+
+        public virtual async Task<ICheckpoint> SetCheckpointAsync(ICheckpoint checkpoint)
+        {
+            EnsureArg.IsNotNull(checkpoint);
+            EnsureArg.IsNotNullOrWhiteSpace(checkpoint.Partition);
+            EnsureArg.IsNotNullOrWhiteSpace(checkpoint.Identifier);
+
+            var lastProcessedDateTime = EnsureArg.IsNotNullOrWhiteSpace(checkpoint.LastProcessedDateTime.DateTime.ToString("MM/dd/yyyy hh:mm:ss.fff tt"), nameof(checkpoint.LastProcessedDateTime));
```

Comment:
`LastProcessedTime` is a `DateTimeOffset` (so struct) and it can't be null. You can probably safely remove this and below just use `checkpoint.LastProcessedDateTime.ToString("MM/dd/yyyy hh:mm:ss.fff tt")`.